File: 백준/Silver/18110. solved．ac/solved．ac.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().trim());

        if(n < 1){
            System.out.println(0);
        }else{
            int[] scores = new int[n];
            for(int i = 0; i < n; i++){
                scores[i] = Integer.parseInt(br.readLine().trim());
            }
            br.close();
    
            Arrays.sort(scores);

            int exclusion = (int) Math.round(n * 0.15);
            int sum = 0;
            for(int i = (0 + exclusion); i < (n - exclusion); i++){
                sum += scores[i];
            }
            int avg = (int) Math.round((double) sum / (double)(n - (2 * exclusion)));

            System.out.println(avg);
        }
    }
}

File: 백준/Silver/18110. solved．ac/solved．ac.py
import sys
import math

input = sys.stdin.readline

def my_round(val):
    if val - int(val) >= 0.5:
        return int(val)+1
    else:
        return int(val)

n = int(input().strip())
if n:
    scores = []

    for _ in range(n):
        scores.append(int(input().strip()))

    scores.sort()

    exclusion = my_round(n * 0.15)

    if exclusion > 0:
        new_scores = scores[exclusion:-exclusion]
    else:
        new_scores = scores

    avg = sum(new_scores) / len(new_scores) if new_scores else 0

    print(my_round(avg))
else:
    print(0)